From 42c5e4b21f114165115bc1527013d011ed25c578 Mon Sep 17 00:00:00 2001
From: Xu Mao <B34688@freescale.com>
Date: Fri, 27 Sep 2013 17:25:35 +0800
Subject: [PATCH 30/70] ENGR00281562 [Android_ MX6Q_SD] : DivxDec: Print crash
 log when play

Divx with Catus player. 100%
When starting player fails and user exit from this activity, onPause
is called and in which pause() and setNormalSpeed() are called, they
are all unnecessary because play state is stopped.

Signed-off-by: Xu Mao <B34688@freescale.com>
---
 .../src/com/freescale/cactusplayer/VideoPlayer.java   | 19 +++++++++++--------
 1 file changed, 11 insertions(+), 8 deletions(-)

diff --git a/CactusPlayer/src/com/freescale/cactusplayer/VideoPlayer.java b/CactusPlayer/src/com/freescale/cactusplayer/VideoPlayer.java
index 04fcfae..8c6c04a 100755
--- a/CactusPlayer/src/com/freescale/cactusplayer/VideoPlayer.java
+++ b/CactusPlayer/src/com/freescale/cactusplayer/VideoPlayer.java
@@ -283,14 +283,17 @@ public class VideoPlayer extends Activity implements SeekBar.OnSeekBarChangeList
 		start();
 	}
 
-	@Override
-	protected void onPause() {
-		Log.d(TAG, CLASS + "onPause");
-		super.onPause();
-		pause();
-        if(mPlaySpeed != 1)
-            setNormalSpeed();
-	}
+    @Override
+    protected void onPause() {
+        Log.d(TAG, CLASS + "onPause");
+        super.onPause();
+        if(mPlayState == PLAYER_PLAYING){
+            pause();
+            Log.d(TAG,"speed is "  + mPlaySpeed);
+            if(mPlaySpeed != 1)
+                setNormalSpeed();
+        }
+    }
 
     @Override
 	protected void onStop() {
-- 
1.8.0

